pkg/sql/show_create.go

```diff
@@ -256,6 +256,11 @@ func (p *planner) ShowCreate(
 		if lErr != nil {
 			return "", lErr
 		}
+		// Overwrite desc with hydrated descriptor.
+		desc, err = lCtx.getTableByID(desc.GetID())
+		if err != nil {
+			return "", err
+		}
```

Comment:
Previously this used to work without this because we were shallow-copying on descriptor construction. Deep-copy makes this necessary.